src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -241,7 +277,7 @@ private static native Point getPoint(long theme, int part, int state,
     public static Point getPoint(String widget, int part, int state, int property) {
         readLock.lock();
         try {
-            return getPoint(getTheme(widget), part, state, property);
+            return getPoint(getTheme(widget, defaultDPI), part, state, property);
```

Comment:
Why does everything else use defaultDPI, is it possible that some values depend on the DPI?

Reply:
> Why does everything else use defaultDPI, is it possible that some values depend on the DPI?

Because Swing's user space coordinates use the default 96dpi. Some values do depend on the DPI, yet layout code still uses the default DPI.

Reply:
What is GC ?

Reply:
> What is GC ?

I guess Graphics Configuration.

> Then that layout code will be broken if layout will be calculated on already visible component, I guess it should take into account the GC where the component is located.

It works fine to the best of Rajat's and my testing because the layout doesn't change. The “icons” don't scale progressively, Windows returns the same size for many DPI values, Swing scales the pixels up, this fact is taken into account when rendering the theme image to avoid additional transformations.

Changing the preferred size dynamically based on the current DPI of the monitor could be a better way in the long term. Dropping `CachedPainter` from the DPI-aware rendering path of the themes is also on the table.

Reply:
We already changed the preferred size dynamically for most of the components, since the size depends on the fonts and the fonts depends on the GC, see:
https://hg.openjdk.org/jdk/jdk11/rev/55a43beaa529
https://hg.openjdk.org/jdk/jdk/rev/ea50ef230511
https://github.com/openjdk/jdk/commit/11aadc9d98d364b91114c028c7e2eff8de2f2bf0